plugins/woocommerce/src/Internal/FraudProtection/PaymentMethodEventTracker.php

```diff
@@ -0,0 +1,260 @@
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\FraudProtection;
+
+use Automattic\WooCommerce\Internal\RegisterHooksInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Tracks payment method events for fraud protection analysis.
+ *
+ * This class hooks into WooCommerce payment method events in My Account
+ * (add, update, set default, delete, add failed) and triggers comprehensive event
+ * tracking with full session context. It orchestrates the event tracking by collecting
+ * session data and preparing it for the fraud protection service.
+ *
+ * @since 10.5.0
+ * @internal This class is part of the internal API and is subject to change without notice.
+ */
+class PaymentMethodEventTracker implements RegisterHooksInterface {
+
+	/**
+	 * Session data collector instance.
+	 *
+	 * @var SessionDataCollector
+	 */
+	private SessionDataCollector $data_collector;
+
+	/**
+	 * Session clearance manager instance.
+	 *
+	 * @var SessionClearanceManager
+	 */
+	private SessionClearanceManager $session_clearance_manager;
+
+	/**
+	 * Fraud protection controller instance.
+	 *
+	 * @var FraudProtectionController
+	 */
+	private FraudProtectionController $fraud_protection_controller;
+
+	/**
+	 * Initialize with dependencies.
+	 *
+	 * @internal
+	 *
+	 * @param SessionDataCollector      $data_collector              The session data collector instance.
+	 * @param SessionClearanceManager   $session_clearance_manager   The session clearance manager instance.
+	 * @param FraudProtectionController $fraud_protection_controller The fraud protection controller instance.
+	 */
+	final public function init(
+		SessionDataCollector $data_collector,
+		SessionClearanceManager $session_clearance_manager,
+		FraudProtectionController $fraud_protection_controller
+	): void {
+		$this->data_collector              = $data_collector;
+		$this->session_clearance_manager   = $session_clearance_manager;
+		$this->fraud_protection_controller = $fraud_protection_controller;
+	}
+
+	/**
+	 * Register payment method event hooks.
+	 *
+	 * Hooks into WooCommerce payment token actions to track fraud protection events.
+	 * Only registers hooks if the fraud protection feature is enabled.
+	 */
+	public function register(): void {
+		// Only register hooks if fraud protection is enabled.
+		if ( ! $this->fraud_protection_controller->feature_is_enabled() ) {
+			return;
+		}
+
+		add_action( 'woocommerce_new_payment_token', array( $this, 'handle_payment_method_added' ), 10, 2 );
+		add_action( 'woocommerce_payment_token_updated', array( $this, 'handle_payment_method_updated' ), 10, 1 );
+		add_action( 'woocommerce_payment_token_set_default', array( $this, 'handle_payment_method_set_default' ), 10, 2 );
+		add_action( 'woocommerce_payment_token_deleted', array( $this, 'handle_payment_method_deleted' ), 10, 2 );
+		add_action( 'woocommerce_payment_token_add_failed', array( $this, 'handle_payment_method_add_failed' ), 10, 2 );
```

Comment:
Payment methods will need to call this action when a payment method fails to be added. We'll need to communicate to them about this so they can take full advantage of the fraud protection feature
Linear issue created: WOOSUBS-1356